### PR TITLE
 [docs/linux] Fix README.Linux.md binary addon installation instructions

### LINKED ISSUE #16293: [docs/linux] README.Linux.md binary addon installation instructions outdated/wrong
### Describe the bug
The `README.Linux.md` describes an in-tree installation of binary addons (such as PVR addons). The in-tree installed addons are not recognized by Kodi, at least when following out-of-tree compilation of Kodi itself as instructed in the Readme. Thus the instructions in the Readme are wrong/misleading. Further out-of-tree install of the addons would be preferable over in-tree install, but is not described in the Readme.

## Expected Behavior
The Readme contains instructions that are simple and actually work.

## Actual Behavior
- Follow Step 1 - 6 of the Readme.Linux.md
- Kodi does not recognize that any binary addons were installed

## Possible Fix
Replace the broken Step 5 In-Tree addon installation instructions with working Out-of-Tree instructions. Alternatively fix the recognition of in-tree installed addons.

Exemplary out-of-tree instructions for the `platform`, `kodi-platform` and `pvr.iptvsimple` are below:

> the platform add-on, required to be able to build&install other add-ons:
> Code:
> 
> cd ~src/
> git clone https://github.com/Pulse-Eight/platform.git
> cd ~src/platform/
> cmake -DCMAKE_INSTALL_PREFIX=/usr/local
> make && make install
> 
> the kodi-platform add-on, required to be able to build&install other add-ons:
> Code:
> 
> cd ~src/
> git clone https://github.com/xbmc/kodi-platform.git
> cd ~src/kodi-platform/
> cmake -DCMAKE_INSTALL_PREFIX=/usr/local
> make && make install
> 
> the pvr.hts add-on, in case you want to connect to a TVheadend backend and watch TV:
> Code:
> 
> cd ~/src
> git clone https://github.com/kodi-pvr/pvr.iptvsimple.git
> cd ~/pvr.iptvsimple/
> cmake -DCMAKE_INSTALL_PREFIX=/usr/local
> make && make install

### To Reproduce
- Follow README.Linux.md Step 1-6

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
- This is a well known problem, that has made trouble for many users. See for example this thread on a linux forum: https://www.linuxquestions.org/questions/slackware-14/kodi-18-pvr-hts-tvheadend-4175647592/
- The instructions for out-of-tree build which I suggest should be incorporated into the README.Linux.md are taken from https://www.linuxquestions.org/questions/slackware-arm-108/kodi-leia-18-x-mediaplayer-optimized-for-raspberry-pi1-zero-pi2b-pi3b-on-slackware-arm-14-2-sf-and-slackware-arm-current-hf-4175647585/#post5957433

### Your Environment
Used Operating system:
 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian Buster (Debian 10)
 - Kodi version: 18.3

### PR DESCRIPTION
**Pull request by leezu**
## Description
Fix https://github.com/xbmc/xbmc/issues/16293

## Motivation and Context

See https://github.com/xbmc/xbmc/issues/16293

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by wsnipex** [comment]
Can you please add a note that the user must provide/install all needed dependencies of the addon(s) themselves.
Needed dependencies can be found by checking the depends folder and it's subdirs of each addon

**Comment by leezu** [comment]
Added https://github.com/leezu/xbmc/commit/0aaaa87efd9a227c65f1aae89e17f76c9a5815a0

File: docs/README.Linux.md [hunk 1]
```diff
@@ -223,6 +232,47 @@ sudo make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons P
 
 **[back to top](#table-of-contents)**
 
+
+### 5.2. Out-of-tree building of binary add-ons
+
+First build the the platform add-on, required to be able to build and install other add-ons:
```
Comment (Rechi) [inline docs/README.Linux.md]: It's not required for all add-ons.
Also in In-tree building https://github.com/xbmc/platform is used.

File: docs/README.Linux.md [hunk 2]
```diff
@@ -223,6 +232,47 @@ sudo make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons P
...
+
+```
+cd ~/src/
+git clone https://github.com/Pulse-Eight/platform.git
+cd ~/src/platform/
+cmake -DCMAKE_INSTALL_PREFIX=/usr/local
+make && make install
+```
+
+Then build the kodi-platform add-on, required to be able to build and install other add-ons:
```
Comment (Rechi) [inline docs/README.Linux.md]: It's not required for all add-ons.

File: docs/README.Linux.md [hunk 3]
```diff
@@ -223,6 +232,47 @@ sudo make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons P
...
+cd ~/src/
+git clone https://github.com/xbmc/kodi-platform.git
+cd ~/src/kodi-platform/
+cmake -DCMAKE_INSTALL_PREFIX=/usr/local
+make && make install
+```
+
+Then install any other addons.
+You can find a complete list of available binary add-ons **[here](https://github.com/xbmc/repo-binary-addons)**.
+Exemplary, to install `pvr.iptvsimple`, follow below steps.
```
Comment (Rechi) [inline docs/README.Linux.md]: better use pvr.demo as demo

File: docs/README.Linux.md [hunk 4]
```diff
@@ -223,6 +232,47 @@ sudo make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons P
...
+For other addons, simply adapt the repository based on the information found in the `.txt` associated with the respective addon **[here](https://github.com/xbmc/repo-binary-addons)**
+
+Some addons may have further dependencies. You must install all needed dependencies of a addon.
```
Comment (Rechi) [inline docs/README.Linux.md]: [...] of an addon.

**Comment by Rechi** [comment]
please squash the commits

**Comment by leezu** [comment]
@Rechi @wsnipex Please clarify where the dependencies are specified. There is no `depends` folder in https://github.com/xbmc/repo-binary-addons.

**Comment by notspiff** [comment]
Required dependencies can be found by checking the depends folder and it's subdirs of the respective addons *git repositories*.

**Comment by leezu** [comment]
@notspiff thanks.

As far as I can see the dependencies are declared in the /respective addons git repositories/ comprehensively only via CMakeList and not in the depends folder.
For example, https://github.com/kodi-pvr/pvr.iptvsimple/tree/Matrix/depends does not declare the `p8-platform` dependency and it is only included in https://github.com/kodi-pvr/pvr.iptvsimple/blob/ee94b98fd74a570eeda8fe98dad176a1882173a2/CMakeLists.txt#L10

**Comment by notspiff** [comment]
You found the one exception and went with it. The platform library is
implicitly part of the build env - it is a kodi addon after all.

The build system naturally probes for the required deps but the depends
folder describes them.

tir. 2. jul. 2019, 12:46 skrev Leonard Lausen <notifications@github.com>:

**Comment by leezu** [comment]
My concern is to document this correctly. Given that the depends folder is not necessarily correct, let's ask people to look at the CMakefile directly. Would that be a reasonable solution?

**Comment by wsnipex** [comment]
that requires a minimum of cmake knowledge. You can point to CMakeLists.txt in addition to the depends folder.

**Comment by leezu** [comment]
Ok

**Comment by wsnipex** [comment]
thanks, looks good to me now.

**Pull request merged**